lib/esp32/espnow.toit

```diff
@@ -0,0 +1,75 @@
+// Copyright (C) 2022 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the lib/LICENSE file.
+
+STATION ::= 0 // Wi-Fi station mode
+SOFTAP  ::= 1 // Not support yet
+
+BROADCAST_MAC ::= #[0xff, 0xff, 0xff, 0xff, 0xff, 0xff]
+
+check_mac_ address/ByteArray:
+  if address.size != 6:
+      throw "ESP-Now MAC address length must be 6 bytes"
+
+check_key_ key/ByteArray:
+  if key.size != 16:
+      throw "ESP-Now key length must be 16 bytes"
+
+check_channel_ channel/int:
+  if channel > 14 or channel < 0:
+      throw "ESP-Now channel range must be 0~14"
+
+class Address:
+  address/ByteArray
```

Comment:
Maybe call this "mac" ?
This way, we don't have the confusion of which "address" we are talking about.
When using "mac" we mean the byte array, when saying "address" we mean the Address class.

Clearly this is a made-up distinction, but it should help when documenting.